### PR TITLE
MAINT: Adjust type promotion in linalg.norm

### PR DESCRIPTION
**Pull request by eric-wieser**
Relates to #10364.

Unfortunately the failing case there still doesn't work when the result is a scalar, due to falling afoul of #10322. We fix this by just not making any promise about the scalar case.

That seems pretty reasonable to me, since numpy makes it very hard to keep float32 scalars around anyway.

### CODE REVIEW

**Comment by charris** [comment]
I added the `backport candidate` label on account of the change to the release notes.

**Review by mhvk — APPROVED** [review]
Looks all OK to me

**Comment by mhvk** [comment]
But the failures of course suggest there is a problem...

**Comment by eric-wieser** [comment]
Failures are caused by #10374

File: numpy/linalg/linalg.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
     add, multiply, sqrt, maximum, fastCopyAndTranspose, sum, isfinite, size,
     finfo, errstate, geterrobj, longdouble, moveaxis, amin, amax, product, abs,
     broadcast, atleast_2d, intp, asanyarray, object_, ones, matmul,
-    swapaxes, divide, count_nonzero
+    swapaxes, divide, count_nonzero, reciprocal
```
Comment (mhvk) [inline numpy/linalg/linalg.py]: Don't need reciprocal any more...

**Comment by mhvk** [comment]
Apart from the nitpick, would seem ready to go in.

**Comment by eric-wieser** [comment]
Updated with the nit fix

**Comment by mhvk** [comment]
OK, will merge now.

**Pull request merged**